src/FedoraApi.php

```diff
@@ -305,4 +307,80 @@ public function getGraph(ResponseInterface $response)
         }
         return $graph;
     }
+
+    /**
+     * Creates version in Fedora.
+     * @param string $uri Fedora Resource URI
+     * @param string $timestamp Timestamp for Memento version
+     * @param string $content String or binary content
+     * @param array $header HTTP Headers
+     *
+     * @return ResponseInterface
+     */
+    public function createVersion(
+        $uri = '',
+        $timestamp = '',
+        $content = null,
+        $headers = []
+    ) {
+        $timemap_uri = $this->getTimemapURI($uri, $headers);
+        $options = ['http_errors' => false];
+        if ($timestamp != ''){
+            $headers['Memento-Datetime'] = $timestamp;
+        }
+        if ($content != null){
+            $options['body'] = $content;
+        }
+        $options['headers'] = $headers;
+
+        return $this->client->request(
+            'POST',
+            $timemap_uri,
+            $options
+        );
+    }
+
+    /**
+     * Gets list of versions in Fedora.
+     * @param string $uri Fedora Resource URI
+     * @param array $header HTTP Headers
+     *
+     * @return ResponseInterface
+     */
+    public function getVersions(
+        $uri = '',
+        $headers = []
+    ) {
+        $timemap_uri = $this->getTimemapURI($uri, $headers);
+        $options = ['http_errors' => false, 'headers' => $headers];
+
+        return $this->client->request(
+            'GET',
+            $timemap_uri,
+            $options
+        );
+    }
+
+    /**
+     * Helper method to get the Headers for a resource
+     * and parse the timemap header from it
+     * @param string $uri Fedora Resource URI
+     * @param array $header HTTP Headers
+     *
+     * @return string
+     */
+    public function getTimemapURI(
+        $uri = '',
+        $headers = []
+    ) {
+        $resource_headers = $this->getResourceHeaders($uri, $headers);
+        $parsed_link_headers = Psr7\parse_header($resource_headers->getHeader('Link'));
+        $timemap_uri = null;
+        $timemap_index = array_search('timemap', array_column($parsed_link_headers, 'rel'));
+        if (is_int($timemap_index)) {
+            $timemap_uri = $parsed_link_headers[$timemap_index][0];
+        }
+        $timemap_uri = trim($timemap_uri, "<> \t\n\r\0\x0B");
```

Comment:
Should this go inside the `if (is_int($timemap_index)) {}` group? I'm wondering if `trim` would have a problem if you tried to trim a NULL?

Reply:
Trim just returns back null when I pass it null, but I agree that it would be better inside the above if statement. d122492